doc/regression-tests.md

```diff
@@ -1,6 +1,102 @@
+
+Having said all that, there is a [primitive mechanism](#checksum-file) which detects changes to the files associated with a given flow since the last time the regression tests ran successfully for that flow.
+
+## Motivation
+
+These tests were introduced by @chrisroos & @jamesmead in 2015 to reduce the risk of doing large-scale refactoring within the application. In particular, they wanted to make a substantial change to the way that landing, question & outcome pages were rendered. The vast majority of the integration tests did not (and still don't) render these pages and in general test coverage was very patchy. The regression tests were created to fulfil this requirement.
+
+The plan is to [refactor all the existing flows](refactoring.md) to separate the "model", "view" & "controller" concerns and use a [new testing approach](new-style-testing.md) which would render the regression tests obsolete.
+
+The regression tests are quite brittle and some of them take a long time to run which can make development a bit painful. However, they are serving an important purpose and we hope to remove them as soon as possible.
+
+## Coverage
+
+At the time of their introduction, Rcov was used to pick a minimal combinations of responses which exercised as much of the logic in the flow as possible. However, coverage is not exhaustive, particularly in flows which use country drop-downs (e.g. Marriage Abroad), or numeric inputs.
+
+Thus it is possible to make a change to the behaviour of a flow and for the regression tests to continue to pass, because you have affected a path that is not covered by the regression tests. In this case, if the new behaviour is significantly different from anything already covered, you should consider adding test coverage for the new behaviour.
+
+However, this new test coverage doesn't have to be at the regression test level, it might be sufficient to do it at the integration test or unit test level.
+
+## Regression test files
+
+### Questions & responses file
+
+This file stores the responses to be used for each question when running the tests. The filenames are of the form: `test/data/<smart-answer-flow-name>-questions-and-responses.yml`.
+
+The `script/generate-questions-and-responses-for-smart-answer.rb` script can be used to generate this file, but it requires manual intervention for questions which don't have a limited/known set of options (i.e. questions other than multi-choice/checkbox questions).
+
+You *will* need to update it if you:
+
+  * add a question
+  * remove a question
+  * rename a question key
+  * remove an option from a multi-choice/checkbox question (if being exercised)
+
+You *might* want to consider updating it (for [coverage purposes](#coverage)) if you:
+
+  * add an option to a multi-choice/checkbox question
+  * change the routing logic
```

Comment:
The (meaning of) term used here ```routing logic```. I feel for a developer beginning with smart answers may vary.

It may further help to define what this means in the smart-answers context.. Up until now I have referred to this as flow logic. having worked on smart-answers.. it is very clear to me what this implies, but the beginner I am not sure what this may imply.